src/systems/filecoin_vm/interpreter/vm_interpreter.go

```diff
@@ -174,32 +180,27 @@ func (vmi *VMInterpreter_I) ApplyMessage(
 		return
 	}
 
-	// make sure this is the right message order for fromActor
-	if message.CallSeqNum() != fromActor.CallSeqNum() {
-		_applyError(inTree, exitcode.InvalidCallSeqNum, SenderResolveSpec_Invalid)
+	// Check receiving actor and method exists, possibly creating an account actor.
+	// If this succeeds, compTreePreSend will become a state snapshot which includes
+	// implicit receiver creation, the sender (rather than miner) paying gas, and the sender's
+	// CallSeqNum being incremented; at least that much state change will be persisted even if the
```

Comment:
What's the rationale for persisting that state?

Reply:
Someone has to pay gas. This marks the transition from the miner being on the hook for a bad message, to the sender for a possibly-ok one.